src/server/scripts/DragonIsles/AberrusTheShadowedCrucible/boss_kazzara_the_hellforged.cpp

```diff
@@ -93,10 +274,448 @@ struct boss_kazzara_the_hellforged : public BossAI
     void UpdateAI(uint32 diff) override
     {
         scheduler.Update(diff);
+
+        if (!UpdateVictim())
+            return;
+
+        if (me->HasUnitState(UNIT_STATE_CASTING))
+            return;
+
+        events.Update(diff);
+
+        if (me->GetPower(me->GetPowerType()) == 100 && !me->HasUnitState(UNIT_STATE_CASTING))
+        {
+            me->RemoveAurasDueToSpell(SPELL_PERIODIC_ENERGIZE);
+            if (Unit* target = SelectTarget(SelectTargetMethod::Random, 0, 70.0f, true))
+            {
+                Talk(SAY_HELLBEAM);
+                Talk(SAY_HELLBEAM_ALERT);
+                DoCast(target, SPELL_HELLBEAM);
+                events.ScheduleEvent(EVENT_ENERGIZE, 1s);
+            }
+        }
+
+        while (uint32 eventId = events.ExecuteEvent())
+        {
+            switch (eventId)
+            {
+                case EVENT_RAYS_OF_ANGUISH:
+                    Talk(SAY_RAYS_OF_ANGUISH);
+                    DoCastSelf(SPELL_RAYS_OF_ANGUISH);
+                    events.ScheduleEvent(EVENT_RAYS_OF_ANGUISH, 34s);
+                    break;
+                case EVENT_TERROR_CLAWS:
+                    Talk(SAY_TERROR_CLAWS);
+                    DoCastVictim(SPELL_TERROR_CLAWS_SELECTOR);
+                    events.ScheduleEvent(EVENT_TERROR_CLAWS, 15s);
+                    break;
+                case EVENT_WINGS_OF_EXTINCTION:
+                    Talk(SAY_WINGS_OF_EXTINCTION);
+                    DoCastSelf(SPELL_WINGS_OF_EXTINCTION);
+                    events.ScheduleEvent(EVENT_WINGS_OF_EXTINCTION, 34s);
+                    break;
+                case EVENT_ENERGIZE:
+                    DoCastSelf(SPELL_PERIODIC_ENERGIZE);
+                    break;
+                case EVENT_DREAD_RIFT:
+                    DoCastSelf(SPELL_DREAD_RIFTS);
+                    events.ScheduleEvent(EVENT_DREAD_RIFT, 34s);
+                default:
+                    break;
+            }
+        }
+
+        DoMeleeAttackIfReady();
+    }
+private:
+    bool _under80Percent;
```

Comment:
You can use a uint8 counter to count the executed carnages instead of using bools

Reply:
Is not too bad for me have the 3 booleans.

Reply:
the point is unifying it into the same variable to keep it cleaner.